dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ProcessingManager.java

```diff
@@ -109,10 +110,13 @@ private ProcessingManager(Configuration config) throws InvalidConfigurationExcep
                 IntrinsicsInlining.fromConfig(config),
                 MemoryAllocation.newInstance(),
                 // --- Statistics + verification ---
-                EventIdReassignment.newInstance(), // Normalize used Ids (remove any gaps)
-                printAfterProcessing ? DebugPrint.withHeader("After processing") : null,
-                CoreCodeVerification.fromConfig(config),
-                LogProgramStatistics.newInstance()
+                IdReassignment.newInstance(), // Normalize used Ids (remove any gaps)
+                printAfterProcessing ? DebugPrint.withHeader("After processing", Printer.Mode.THREADS) : null,
```

Comment:
Is this intended to replace #491 ?

Reply:
Partly yes. We still might want to delete functions if they end up consuming too much memory, but just deleting them for the sake of better printing seemed a little overkill, so I made the `Printer` more smart.